### PR TITLE
Identify cancelled tasks in list tasks API

### LINKED ISSUE #72907: Show that cancellable task has been cancelled in tasks API
Task cancellation works by having the task poll the cancellation flag whenever it is convenient to do so. In some cases the convenient times might be separated by a very long interval, so even if the task is cancelled it still runs for a while and appears in the output of the tasks API. Today we don't expose the cancellation flag in the tasks API so users cannot tell whether any given task has already been cancelled, which can be frustrating.

The tasks API should include a `"cancelled": true` flag on tasks that have been cancelled.

Splitting this idea out from the more general https://github.com/elastic/elasticsearch/issues/67505 to clarify the action needed.

### PR DESCRIPTION
**Pull request by DaveCTurner**
This commit adds a `cancelled` flag to each cancellable task in the
response to the list tasks API, allowing users to see that a task has
been properly cancelled and will complete as soon as possible.

Closes #72907

### CODE REVIEW

File: docs/reference/cluster/tasks.asciidoc [hunk 1]
```diff
@@ -243,6 +244,13 @@ nodes `nodeId1` and `nodeId2`.
 POST _tasks/_cancel?nodes=nodeId1,nodeId2&actions=*reindex
 --------------------------------------------------
 
+A task may continue to run for some time after it has been cancelled because it
```
Comment (fcofdez) [inline docs/reference/cluster/tasks.asciidoc:247]: 👍

**Review by fcofdez — APPROVED** [review]
LGTM, thanks David!

**Review by tlrx — APPROVED** [review]
LGTM, this is the kind of small changes that makes everyone's life easier :+1:

**Comment by DaveCTurner** [comment]
Thanks both!

**Pull request merged**